### ISSUE TITLE
Game gets stuck on building terrain when loading a world after restarting 

### ISSUE THREAD

**Issue by ikeigher** [DarkPacks/SevTech-Ages#4727]
## Bug Description

When I create a fresh world I can enter and play as normal. However, after I save and close the game, the next time I try to open the world It will get stuck on "Loading World, Building Terrain". It also makes my computer not work correctly when I close it with ALT+F4 because it will not close any other way, this means I have to hard restart my PC each time.

### Did this behavior use to work in the previous version?

I was able to play fine without this issue before 3.2.3

## Steps to Reproduce (for bugs)

1. Create a fresh new world and play for any amount of time.
2. Save and close the world and then restart the game.
3. Attempt to load the previously used world.

## Logs

* Client/Server Log: https://gist.github.com/ikeigher/8bede3e74e3bc7ee6a00081b99c0f174
* Crash Log:

## Client Information
* Modpack Version: 3.2.3
* Java Version: 1.8.0 341
* Launcher Used: Curseforge - Forge 2860
* Memory Allocated: 8192
* Server/LAN/Single Player: Single Player
* Optifine Installed: No
* Shaders Enabled: No

## World Information
* Modpack Version world created in: 3.2.3
* Additional Content Installed: None

**Comment by sam-kirby** [comment]
You have half the amount of memory assinged compared to what you think [`-Xmx4096m`](https://gist.github.com/ikeigher/8bede3e74e3bc7ee6a00081b99c0f174#file-gistfile1-txt-L410). This is probably a bit low, so try upping it to at least 6GB.

Otherwise your log looks normal and the world is loaded properly by the internal server.

**Comment by ikeigher** [comment]
> You have half the amount of memory assinged compared to what you think [`-Xmx4096m`](https://gist.github.com/ikeigher/8bede3e74e3bc7ee6a00081b99c0f174#file-gistfile1-txt-L410). This is probably a bit low, so try upping it to at least 6GB.
> 
> Otherwise your log looks normal and the world is loaded properly by the internal server.

Should I do that in the curseforge settings before the launcher or change it on the launcher. Or does it not matter which really.
Ill give it a try now though because it could be resetting each time.

**Comment by sam-kirby** [comment]
> Should I do that in the curseforge settings before the launcher or change it on the launcher. Or does it not matter which really.

You need to do it in Curseforge as the launcher gets reset frequently.

**Comment by ikeigher** [comment]
Alright I tried increasing the amount to 6144MB and 8192MB on the curseforge launcher and sadly no luck there. I'll keep trying different things

**Comment by ikeigher** [comment]
I haven't had any luck so far, I'm at a bit of a loss now

**Comment by sam-kirby** [comment]
I've not heard of this happening before, but you might want to try dividing the labour by hosting the world using the dedicated server.

With 16GB of RAM you should be able to do this on the same computer without issue.

**Comment by ikeigher** [comment]
Ah alright, I did notice it worked whenever I joined another server but just not single player. I'll give that a go and get back to you. Thanks

**Comment by ikeigher** [comment]
I seem to be having the same problem with a dedicated server, I followed all the steps and after restarting it doesn't finish loading or at least it looks to be that way.

**Comment by ikeigher** [comment]
It says it is unloading dimension 53 which is the last thing in the console but I think there is supposed to be a few more unloaded dimension lines after that, there is at least when I initially start up the server

**Comment by sam-kirby** [comment]
As far as I can tell from the source, dimensions are only unloaded once the game has started ticking. This means that the server should be accepting connections - have you actually tried connecting to it? What log output are you expecting to come after?

**Comment by ikeigher** [comment]
Yeah when I first launch the server it allows me to connect with my IPv4 on direct connect and the world is all good and with that it also loads these lines after dimension 53 too:
[20:01:25] [Server thread/INFO] [FML]: Unloading dimension 144
[20:01:25] [Server thread/INFO] [FML]: Unloading dimension 28885
[20:01:25] [Server thread/INFO] [FML]: Unloading dimension 20
[20:01:25] [Server thread/INFO] [FML]: Unloading dimension 7
[20:01:25] [Server thread/INFO] [FML]: Unloading dimension -1
[20:01:25] [Server thread/INFO] [FML]: Unloading dimension 10
[20:01:25] [Server thread/INFO] [FML]: Unloading dimension 1
[20:02:09] [Server thread/INFO] [antiqueatlas]: Saving local markers data to NBT

The next time these don't show up and it just says logging in when I attempt to join with the IP.

**Comment by sam-kirby** [comment]
Huh, well I'm stumped, and unless you want to learn how to use VisualVM I'm not really sure what to suggest next.

I cannot replicate this no matter what I try. One thing you could investigate is hardware issues; have you tried putting the game on a different storage device?

**Comment by ikeigher** [comment]
Oh no I haven't. I have a few different drives so I'll try all of them to be sure.

**Comment by ikeigher** [comment]
None of them seem to solve the issue when I installed it in different places. It must be something else to do with my system maybe but I cant think of anything that may have broken it.
